aci/resource_aci_vzsubj.go

```diff
@@ -353,6 +483,42 @@ func resourceAciContractSubjectUpdate(ctx context.Context, d *schema.ResourceDat
 		return diag.FromErr(err)
 	}
 
+	if d.HasChange("apply_both_directions") || d.HasChange("consumer_to_provider") || d.HasChange("provider_to_consumer") {
```

Comment:
Don't think you need d.HasChange("apply_both_directions") anymore.

Reply:
if apply_both_directions == "no" && (d.HasChange("consumer_to_provider") || d.HasChange("provider_to_consumer"))